service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/C35aSupervisionOrderDocumentParameterGeneratorTest.java

```diff
@@ -0,0 +1,279 @@
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_CALENDAR_DAY;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_CALENDAR_DAY_AND_TIME;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_NUMBER_OF_MONTHS;
+import static uk.gov.hmcts.reform.fpl.model.order.Order.C35A_SUPERVISION_ORDER;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE_TIME_WITH_ORDINAL_SUFFIX;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE_WITH_ORDINAL_SUFFIX;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateTimeBaseUsingFormat;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.getDayOfMonthSuffix;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith({MockitoExtension.class})
+class C35aSupervisionOrderDocumentParameterGeneratorTest {
+    private static final Time time = new FixedTimeConfiguration().stoppedTime();
+    private static final String CHILD_GRAMMAR = "child";
+    private static final String CHILDREN_GRAMMAR = "children";
+    private static final String LA_CODE = "LA_CODE";
+    private static final String LA_NAME = "Local Authority Name";
+    private static final Child CHILD = mock(Child.class);
+    private static final GeneratedOrderType TYPE = GeneratedOrderType.SUPERVISION_ORDER;
+    private static final String FURTHER_DIRECTIONS = "further directions";
+    private static final LocalDateTime NEXT_WEEK_DATE_TIME = time.now().plusDays(7);
+    private String dayOrdinalSuffix;
+
+    @Mock
+    private ChildrenService childrenService;
+
+    @Mock
+    private LocalAuthorityNameLookupConfiguration laNameLookup;
+
+    @InjectMocks
+    private C35aSupervisionOrderDocumentParameterGenerator underTest;
+
+    @Test
+    void accept() {
+        assertThat(underTest.accept()).isEqualTo(Order.C35A_SUPERVISION_ORDER);
+    }
+
+    @Test
+    void template() {
+        assertThat(underTest.template()).isEqualTo(DocmosisTemplates.ORDER);
+    }
+
+    @Test
+    void shouldReturnContentForSingleChildAndSpecifiedDate() {
+        CaseData caseData = buildCaseDataWithDateSpecified();
+        dayOrdinalSuffix = getDayOfMonthSuffix(NEXT_WEEK_DATE_TIME.getDayOfMonth());
+        String formattedDate = formatLocalDateTimeBaseUsingFormat(
+            NEXT_WEEK_DATE_TIME,
+            String.format(DATE_WITH_ORDINAL_SUFFIX, dayOrdinalSuffix)
+        );
+        String courtOrderMessage = "The Court orders " + LA_NAME
+            + " supervises the " + CHILDREN_GRAMMAR
+            + " until " + formattedDate + ".";
+
+        when(laNameLookup.getLocalAuthorityName(LA_CODE)).thenReturn(LA_NAME);
+
+        List<Element<Child>> selectedChildren = wrapElements(CHILD, CHILD);
```

Comment:
this test title is for a single child, but it looks like you've created a scenario with multiple children. should this be the following instead?
```suggestion
        List<Element<Child>> selectedChildren = wrapElements(CHILD);
```

Reply:
You're 100% correct, well spotted. Ill update it in the next commit that should retrigger its build

Reply:
updated